### PR TITLE
chore: group all non dhis2 deps into a single dependabot PR

### PR DESCRIPTION
**Pull request by jenniferarnesen**
There is a lot of customization we can do with dependabot. To start with, this change will instruct dependabot to group all dev dependencies into a single PR (minor and patch). major will still be a separate PR. It also excludes all dev dependencies from @dhis2

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
       schedule:
-          interval: 'weekly'
+          interval: weekly
       open-pull-requests-limit: 5
-      versioning-strategy: 'increase'
+      versioning-strategy: increase
+      groups:
+          dev-dependencies:
+              applies-to: version-updates
+              dependency-type: development
```
Comment (HendrikThePendric) [inline .github/dependabot.yml:12]: I would personally be open to / in favour of actually removing this line and do this for regular dependencies too. We already exclude security fixes and everything from `@dhis2` and I don't think we're in such a hurry with all the other stuff.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+              update-types:
+                  - minor
+                  - patch
```
Comment (HendrikThePendric) [inline .github/dependabot.yml:15]: I agree that excluding `major` here is a good choice. When grouping things together it becomes harder to track down regressions to an individual lib so excluding version bumps with breaking changes makes sense.

**Review by HendrikThePendric — COMMENTED** [review]
Great stuff @jennifer, this wasn't possible a few years ago and I expect this to be a small job that can make our lives _a lot easier_.

I've suggested one change but would be happy if this is merged in it's current form too.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+      directory: /
       schedule:
-          interval: 'weekly'
+          interval: weekly
       open-pull-requests-limit: 5
-      versioning-strategy: 'increase'
+      versioning-strategy: increase
+      groups:
+          dev-dependencies:
+              applies-to: version-updates
```
[the remark this answers is not in the document]
Comment (jenniferarnesen) [reply .github/dependabot.yml:11]: So, you are you saying to leave line 11 as is?

[the remark this answers is not in the document]
Comment (HendrikThePendric) [reply .github/dependabot.yml:11]: Yes! I sort of expected others to perhaps also provide input and wanted to emphasise that I think this is a good choice 👍

**Pull request merged**